doc/designs/edns.md

```diff
@@ -0,0 +1,141 @@
+
+## Implementation
+
+Most changes involve configuration updates. New installation options for encrypted DNS ensure the deployment and activation of DNS services to encrypt all outbound traffic.
+
+Address of our unbound server has to be set in /etc/resolv.conf. /etc/resolv.conf nameserver 127.0.0.1
+
+
+## How to Use
+During the deployment of a FreeIPA server, replica, or client, new options are available to enable DoT support. These options allow administrators to enhance the security of DNS traffic. Here’s how to use the new options:
+
+- Enable DoT: use the `--dns-over-tls` option to enable DoT support during the deployment of clients, servers, or replicas. This option deploys Unbound as a local cache resolver and configures BIND on the server to receive DoT requests.
+
+- Specify an Upstream DNS Server with DoT enabled: use the `--dot-forwarder` option to specify the upstream DNS server that supports DoT. The format must be 1.2.3.4#dns.server.test. You still need to specify at least one of --forwarder, --auto-forwarders, or --no-forwarders options for the non-encrypted communication, discovery process, etc..
+
+- DoT Certificates. If you prefer to use a self-generated certificate for DoT in BIND/Unbound, use the `--dns-over-tls-key` and `--dns-over-tls-cert` options. If these options are not specified, the IPA CA will be used to request a new certificate.
+
+
+## Feature Management
+
+### UI
+
+TBD.
+
+### CLI
+
+Overview of the CLI commands for the FreeIPA installers and FreeIPA DNS installers:
+
+Configuring DNS services using ipa-dns-install follows the same principles as installing DNS with the ipa-server-install utility.
+
+
+| Option | Description                                       |
+|:------------------------ | :------------------------------ |
+|    --dns-over-tls | enable DNS over TLS support. This option is present on both client and server. It deploys Unbound and configures BIND on the server to receive DoT requests. |
+    --dot-forwarder | the upstream DNS server with DoT support. It must be specified in the format 1.2.3.4#dns.server.test |
+    --dns-over-tls-key and --dns-over-tls-cert | in case user prefers to have the DoT certificate in BIND generated by themselves. If these are empty, IPA CA is used instead to request a new certificate. |
+
+
+
+### Configuration
+
+Any configuration options? Any commands to enable/disable the feature or turn on/off its parts? 
```

Comment:
Installers will deploy a valid configuration, but can be easily edited at:` /etc/unbound/conf.d/zzz-ipa.conf`

Reply:
added